save-common/src/commonMain/kotlin/org/cqfn/save/core/config/TestConfig.kt

```diff
@@ -71,50 +73,36 @@ data class TestConfig(
     fun getAllTestConfigs(): List<TestConfig> {
         return listOf(this) + this.childConfigs.flatMap { it.getAllTestConfigs() }
     }
-    
+
     /**
-     * Merge parent configurations with current and prolong it for all child configs
+     * filtering out general configs
      */
-    fun merge() {
-        logDebug("Start merging configs for ${this.location}")
-        val parentConfigs = parentConfigs(withSelf = true).toList().asReversed()
-        mergeConfigList(parentConfigs)
-        mergeChildConfigs()
-    }
+    fun pluginConfigsWithoutGeneralConfig() = pluginConfigs.filterNot { it is GeneralConfig }
 
-    // Merge list of configs pairwise
-    private fun mergeConfigList(configList: List<TestConfig>) {
-        if (configList.size == 1) {
-            return
-        }
-        val pairs = configList.zipWithNext()
-
-        pairs.forEach { (parent, child) ->
-            child.mergeChildConfigWithParent(parent)
-        }
-    }
+    /**
+     * Merge parent list of plugins with the current list
+     *
+     * @return merged test config
+     */
+    fun mergeConfigWithParent(): TestConfig {
+        logDebug("Start merging configs for ${this.location}")
 
-    // Merge child configs recursively
-    private fun mergeChildConfigs() {
-        for (child in childConfigs) {
-            child.mergeChildConfigWithParent(parent = this)
-            child.mergeChildConfigs()
+        // return from the function if we stay at the root element of the plugin tree
+        val parentalPlugins = this.parentConfig?.pluginConfigs ?: return this
+        parentalPlugins.forEach { currentConfig ->
+            val childConfigs = this.pluginConfigs.filter { it.type == currentConfig.type }
```

Comment:
Can we compare `it::class` and `currentConfig::class` in common code?

Reply:
discussed - these are problems of future us